src/bun.js/test/jest.zig

```diff
@@ -2771,7 +2771,94 @@ pub const Expect = struct {
 
         const expected_line = "Expected constructor: <green>{any}<r>\n";
         const received_line = "Received value: <red>{any}<r>\n";
-        const fmt = comptime getSignature("toBeInstanceOf", "", false) ++ "\n\n" ++ expected_line ++ received_line;
+        const fmt = comptime getSignature("toBeInstanceOf", "<green>expected<r>", false) ++ "\n\n" ++ expected_line ++ received_line;
+        if (Output.enable_ansi_colors) {
+            globalObject.throw(Output.prettyFmt(fmt, true), .{ expected_fmt, value_fmt });
+            return .zero;
+        }
+
+        globalObject.throw(Output.prettyFmt(fmt, false), .{ expected_fmt, value_fmt });
+        return .zero;
+    }
+
+    pub fn toMatch(this: *Expect, globalObject: *JSC.JSGlobalObject, callFrame: *JSC.CallFrame) callconv(.C) JSValue {
+        defer this.postMatch(globalObject);
+
+        const thisValue = callFrame.this();
+        const _arguments = callFrame.arguments(1);
+        const arguments: []const JSValue = _arguments.ptr[0.._arguments.len];
+
+        if (arguments.len < 1) {
+            globalObject.throwInvalidArguments("toMatch() requires 1 argument", .{});
+            return .zero;
+        }
+
+        if (this.scope.tests.items.len <= this.test_id) {
+            globalObject.throw("toMatch() must be called in a test", .{});
+            return .zero;
+        }
+
+        active_test_expectation_counter.actual += 1;
+
+        var formatter = JSC.ZigConsoleClient.Formatter{ .globalThis = globalObject, .quote_strings = true };
+
+        const expected_value = arguments[0];
+        if (!expected_value.isString() and !expected_value.isRegExp()) {
+            globalObject.throw("Expected value must be a string or regular expression: {any}", .{expected_value.toFmt(globalObject, &formatter)});
+            return .zero;
+        }
+        expected_value.ensureStillAlive();
+
+        const value = Expect.capturedValueGetCached(thisValue) orelse {
+            globalObject.throw("Internal consistency error: the expect(value) was garbage collected but it should not have been!", .{});
+            return .zero;
+        };
+        value.ensureStillAlive();
+
+        if (!value.isString()) {
+            globalObject.throw("Received value must be a string: {any}", .{value.toFmt(globalObject, &formatter)});
+            return .zero;
+        }
+
+        const not = this.op.contains(.not);
+        var pass: bool = brk: {
+            if (expected_value.isString()) {
+                if (value.getPrototype(globalObject).get(globalObject, "includes")) |includes_fn| {
+                    const result = includes_fn.callWithThis(globalObject, value, &.{expected_value});
+                    break :brk result.toBoolean();
+                }
+            } else if (expected_value.isRegExp()) {
+                if (expected_value.get(globalObject, "test")) |test_fn| {
+                    const result = test_fn.callWithThis(globalObject, expected_value, &.{value});
```

Comment:
we need to check that `test_fn` doesn't pass `isEmptyOrUndefinedOrNull()`. It's unlikely but `RegExp.prototype.test = undefined` will cause the tests to segfault.

Reply:
This should be a binding, we should avoid calling into JS functions. Malicious user could implement their own function called `"test"`, it might not be a function, etc.

Something like this on `JSValue`:
```zig
expected_value.toMatch(globalObject, other: JSValue)
```

On the C++ binding, we can call `matchInline`

https://github.com/oven-sh/WebKit/blob/bb47d871bd5c130a72f1dd827cb1d5d06ed9832c/Source/JavaScriptCore/runtime/RegExpObjectInlines.h#L97-L99

```c++
inline MatchResult RegExpObject::matchInline(
    JSGlobalObject* globalObject, JSString* string)
{
```

The type is `JSC::RegExpObject`